pkg/cli/cli.go

```diff
@@ -191,90 +191,94 @@ func (c *cli) getInfoFromFlags() (string, []string, error) {
 // the config file is valid and uses default values in case some info was not provided.
 func (c cli) resolveFlagsAndConfigFileConflicts(
-	flagProjectVersion, cfgProjectVersion string,
+	flagProjectVersionString string,
+	cfgProjectVersion config.Version,
 	flagPlugins, cfgPlugins []string,
-) (string, []string, error) {
+) (config.Version, []string, error) {
+	// Parse project configuration version from flags
+	var flagProjectVersion config.Version
+	if flagProjectVersionString != "" {
+		if err := flagProjectVersion.Parse(flagProjectVersionString); err != nil {
+			return config.Version{}, nil, fmt.Errorf("unable to parse project version flag: %w", err)
+		}
+	}
+
 	// Resolve project version
-	var projectVersion string
+	var projectVersion config.Version
 	switch {
-	// If they are both blank, use the default
-	case flagProjectVersion == "" && cfgProjectVersion == "":
+	// If they are both invalid (empty is invalid), use the default
+	case flagProjectVersion.Validate() != nil && cfgProjectVersion.Validate() != nil:
 		projectVersion = c.defaultProjectVersion
-	// If they are equal doesn't matter which we choose
-	case flagProjectVersion == cfgProjectVersion:
+	// If any is invalid (empty is invalid), choose the other
+	case cfgProjectVersion.Validate() != nil:
 		projectVersion = flagProjectVersion
-	// If any is blank, choose the other
-	case cfgProjectVersion == "":
-		projectVersion = flagProjectVersion
-	case flagProjectVersion == "":
+	case flagProjectVersion.Validate() != nil:
 		projectVersion = cfgProjectVersion
-	// If none is blank and they are different error out
+	// If they are equal doesn't matter which we choose
+	case flagProjectVersion.Compare(cfgProjectVersion) == 0:
+		projectVersion = flagProjectVersion
+	// If both are valid (empty is invalid) and they are different error out
 	default:
-		return "", nil, fmt.Errorf("project version conflict between command line args (%s) "+
-			"and project configuration file (%s)", flagProjectVersion, cfgProjectVersion)
-	}
-	// It still may be empty if default, flag and config project versions are empty
-	if projectVersion != "" {
-		// Validate the project version
-		if err := validation.ValidateProjectVersion(projectVersion); err != nil {
-			return "", nil, err
-		}
+		return config.Version{}, nil, fmt.Errorf("project version conflict between command line args (%s) "+
+			"and project configuration file (%s)", flagProjectVersionString, cfgProjectVersion)
 	}
 
 	// Resolve plugins
 	var plugins []string
 	switch {
-	// If they are both blank, use the default
+	// If they are both empty, use the default
 	case len(flagPlugins) == 0 && len(cfgPlugins) == 0:
-		plugins = c.defaultPlugins[projectVersion]
-	// If they are equal doesn't matter which we choose
-	case equalStringSlice(flagPlugins, cfgPlugins):
-		plugins = flagPlugins
-	// If any is blank, choose the other
+		if defaults, hasDefaults := c.defaultPlugins[projectVersion]; hasDefaults {
```

Comment:
If there are no default plugins for a project version the plugin keys will be empty, resolving into a empty list of resolved plugins (obviously) and when binding the different plugins to their commands, they will not find any API plugin and so on so it will build a command that errors out.

I haven't changed this behavior, if we wish to change this I think we should discuss it in the EP in #1942